packages/core/src/amazonq/webview/ui/apps/BaseConnector.ts

```diff
@@ -0,0 +1,298 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ChatItem, ChatItemAction, ChatItemType, FeedbackPayload } from '@aws/mynah-ui'
+import { ExtensionMessage } from '../commands'
+import { CodeReference } from './amazonqCommonsConnector'
+import { TabOpenType, TabsStorage, TabType } from '../storages/tabsStorage'
+import { FollowUpGenerator } from '../followUps/generator'
+import { CWCChatItem } from '../connector'
+
+interface ChatPayload {
+    chatMessage: string
+    chatCommand?: string
+}
+
+export interface BaseConnectorProps {
+    sendMessageToExtension: (message: ExtensionMessage) => void
+    onMessageReceived?: (tabID: string, messageData: any, needToShowAPIDocsTab: boolean) => void
+    onChatAnswerReceived?: (tabID: string, message: CWCChatItem | ChatItem, messageData: any) => void
+    onError: (tabID: string, message: string, title: string) => void
+    onWarning: (tabID: string, message: string, title: string) => void
+    onOpenSettingsMessage: (tabID: string) => void
+    tabsStorage: TabsStorage
+}
+
+export abstract class BaseConnector {
+    protected readonly sendMessageToExtension
+    protected readonly onError
+    protected readonly onWarning
+    protected readonly onChatAnswerReceived
+    protected readonly onOpenSettingsMessage
+    protected readonly followUpGenerator: FollowUpGenerator
+    protected readonly tabsStorage
+
+    abstract getTabType(): TabType
+
+    constructor(props: BaseConnectorProps) {
+        this.sendMessageToExtension = props.sendMessageToExtension
+        this.onChatAnswerReceived = props.onChatAnswerReceived
+        this.onWarning = props.onWarning
+        this.onError = props.onError
+        this.onOpenSettingsMessage = props.onOpenSettingsMessage
+        this.tabsStorage = props.tabsStorage
+        this.followUpGenerator = new FollowUpGenerator()
+    }
+
+    onResponseBodyLinkClick = (tabID: string, messageId: string, link: string): void => {
+        this.sendMessageToExtension({
+            command: 'response-body-link-click',
+            tabID,
+            messageId,
+            link,
+            tabType: this.getTabType(),
+        })
+    }
+    onInfoLinkClick = (tabID: string, link: string): void => {
+        this.sendMessageToExtension({
+            command: 'footer-info-link-click',
+            tabID,
+            link,
+            tabType: this.getTabType(),
+        })
+    }
+
+    followUpClicked = (tabID: string, messageId: string, followUp: ChatItemAction): void => {
+        /**
+         * We've pressed on a followup button and should start watching that round trip telemetry
+         */
+        this.sendMessageToExtension({
```

Comment:
I'd probably move these out of here for now and into the cwchatconnector. I haven't checked whether or not the entire chat round trip telemetry flow works for gumby/feature dev yet

Reply:
Discussed Offline - `amazonq_chatRoundTrip` is not getting emitted for `/dev`, so this is good to go.